fc/cli/reports.py

```diff
@@ -0,0 +1,503 @@
+    cli_library.echo('Generating user tasks to google sheet')
+
+    auth = ComboAuth(username)
+
+    # setup and build credentials for google api calls
+    g_scopes = None
+    g_service_acct_file = None
+    g_sheet_create_url = None
+
+    (g_scopes, g_service_acct_file, g_sheet_create_url) = build_google_creds(auth)
+
+    if g_service_acct_file is None or g_sheet_create_url is None:
+        cli_library.echo('Google service account credential file path and sheet create url must be defined '
+                         'in order to generate a report file.')
+        return
+
+    credentials = service_account.Credentials.from_service_account_file(
+        g_service_acct_file, scopes=g_scopes)
+
+    service = build('sheets', 'v4', credentials=credentials)
+
+    report_title = 'User Task Report {}'.format(date.today().strftime('%m-%d-%y'))
+
+    response = requests.get(g_sheet_create_url + report_title)
+    response.raise_for_status()
+
+    response_json = response.json()
+
+    spreadsheet_id = None
+    folder_id = None
+
+    if response_json['success']:
+        spreadsheet_id = response_json['newSheetId']
+        folder_id = response_json['destFolderId']
+        cli_library.echo('successful spreadsheet creation, https://docs.google.com/spreadsheets/d/{}/edit#gid=0, '
+                         .format(spreadsheet_id) +
+                         'folder: {}'.format(folder_id))
+
+    # update the individual sheet name in the spreadsheet we just created
+    sheet_update = {
+        'requests': {
+            'updateSheetProperties': {
+                'properties': {
+                    'title': 'Names'
+                },
+                'fields': 'title'
+            }
+        }
+    }
+
+    response = service.spreadsheets().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=sheet_update).execute()
+
+    cli_library.echo('first sheet title updated')
+
+    # add data header row for names sheet
+    # get users
+    # iterate through users
+    #   add data row to name sheet with name
+    #   create an add sheet request for the user
+    #   create a data header row for user sheet
+    #   get tasks for user
+    #     add data row to user sheet for each task: key (link), issue type, summary, status, vfr, due date, importance
+    #     , loe
+    # execute batch update for adding all sheets
+    # execute batch update for adding all data
+
+    value_data = []
+    add_sheet_requests = []
+    add_sheet = {
+        'requests': add_sheet_requests
+    }
+    data_body = {
+        'valueInputOption': 'USER_ENTERED',
+        'data': value_data
+    }
+
+    name_sheet_row_idx = 1
+
+    value_data.append({'range': 'A{}:B{}'.format(name_sheet_row_idx, name_sheet_row_idx),
+                       'values': [['Name', 'EUA']]})
+    name_sheet_row_idx += 1
+
+    dev_user_json = tasks.get_developer_users(auth)
+
+    user_sheet_row_idx = 1
+    user_tasks_json = None
+
+    for user in dev_user_json['actors']:
```

Comment:
Looks like there are three sections of code starting here and extending below.  One for user assigned tasks, unassigned and in progress tasks, and unassigned and open tasks.  They seem roughly separated by a bunch of `#`.  Each of these sections could be put into its own function.